specification/search/data-plane/Azure.Search/preview/2023-07-01-Preview/searchservice.json

```diff
@@ -0,0 +1,11599 @@
+          "description": "Defines the title field to be used for semantic ranking, captions, highlights, and answers. If you don't have a title field in your index, leave this blank."
+        },
+        "prioritizedContentFields": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SemanticField"
+          },
+          "description": "Defines the content fields to be used for semantic ranking, captions, highlights, and answers. For the best result, the selected fields should contain text in natural language form. The order of the fields in the array represents their priority. Fields with lower priority may get truncated if the content is long."
+        },
+        "prioritizedKeywordsFields": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SemanticField"
+          },
+          "description": "Defines the keyword fields to be used for semantic ranking, captions, highlights, and answers. For the best result, the selected fields should contain a list of keywords. The order of the fields in the array represents their priority. Fields with lower priority may get truncated if the content is long."
+        }
+      },
+      "externalDocs": {
+        "url": "https://docs.microsoft.com/azure/search/semantic-search-overview"
+      },
+      "description": "Describes the title, content, and keywords fields to be used for semantic ranking, captions, highlights, and answers."
+    },
+    "SemanticField": {
+      "properties": {
+        "fieldName": {
+          "type": "string",
+          "description": "",
+          "x-nullable": false
+        }
+      },
+      "description": "A field that is used as part of the semantic configuration."
+    },
+    "VectorSearch": {
+      "properties": {
+        "algorithmConfigurations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VectorSearchAlgorithmConfiguration"
+          },
+          "description": "Contains configuration options specific to the algorithm used during indexing time."
+        }
+      },
+      "description": "Contains configuration options related to vector search."
+    },
+    "VectorSearchAlgorithmConfiguration": {
+      "properties": {
+        "name": {
+          "externalDocs": {
+            "url": "https://docs.microsoft.com/rest/api/searchservice/Naming-rules"
+          },
+          "type": "string",
+          "description": "The name to associate with this particular configuration.",
+          "x-nullable": false
+        },
+        "kind": {
+          "type": "string",
+          "description": "The name of the kind of algorithm being configured for use with vector search. Only `hnsw` is supported in the current preview.",
+          "x-nullable": false
+        },
+        "hnswParameters": {
+          "$ref": "#/definitions/HnswParameters",
+          "description": "Contains the parameters specific to hnsw algorithm."
+        }
```

Comment:
Should `kind` be a discriminator, and `hnswParameters` be in the model inherit `VectorSearchAlgorithmConfiguration` (so that other type/kind with different parameters can be added in future)?

This need to be solved before GA. Preview is fine.

Reply:
I think there have been extensive discussions around this already, and we are opting not to go this route, but I could be wrong. Regardless, as you mentioned we can move forward as is for preview.

Reply:
Yes, Weidong is correct here.  I think this was accidentally flattened into the base class, but we can fix that before the next release.